src/api/procedures/setVenueFiltering.ts

```diff
@@ -0,0 +1,62 @@
+import { Asset, PolymeshError, Procedure } from '~/internal';
+import { ErrorCode, SetVenueFilteringParams, TxTags } from '~/types';
+import { ExtrinsicParams, ProcedureAuthorization, TransactionSpec } from '~/types/internal';
+
+/**
+ * @hidden
+ */
+export type Params = SetVenueFilteringParams;
+
+/**
+ * @hidden
+ */
+export async function prepareVenueFiltering(
+  this: Procedure<Params, void>,
+  args: Params
+): Promise<TransactionSpec<void, ExtrinsicParams<'settlement', 'setVenueFiltering'>>> {
+  const {
+    context: {
+      polymeshApi: { tx, query },
+    },
+  } = this;
+
+  const { ticker, enabled: setEnabled } = args;
+
+  const isEnabled = await query.settlement.venueFiltering(ticker);
+
+  console.log({ isEnabled, setEnabled });
+
+  if (isEnabled.valueOf() === setEnabled) {
+    throw new PolymeshError({
+      code: ErrorCode.NoDataChange,
+      message: `Venue filtering is already ${setEnabled ? 'enabled' : 'disabled'}`,
+    });
+  }
+
+  return {
+    transaction: tx.settlement.setVenueFiltering,
+    args: [ticker, setEnabled],
```

Comment:
Instead of directly passing `ticker` and `setEnabled` you need to first convert them to specific chain types using some existing conversion functions. `stringToTicker` for ticker and `booleanToBool` for setEnabled